ext/native/thin3d/VulkanQueueRunner.cpp

```diff
@@ -422,6 +428,66 @@ void VulkanQueueRunner::RunSteps(VkCommandBuffer cmd, const std::vector<VKRStep
 	}
 }
 
+void VulkanQueueRunner::ApplyMGSHack(std::vector<VKRStep *> &steps) {
+	// We want to turn a sequence of copy,render(1),copy,render(1),copy,render(1) to copy,copy,copy,render(n).
```

Comment:
In theory, since we render bind with UV, we may be able to detect if these copies are non-overlapping and do this in a general case, right?

-[Unknown]

Reply:
Yeah, it can definitely be done, might at some point. Not sure how many games benefit though, and it's more work :)